dot/sync/chain_processor_integration_test.go

```diff
@@ -13,14 +13,88 @@ import (
 	"github.com/ChainSafe/gossamer/dot/network"
 	"github.com/ChainSafe/gossamer/dot/state"
 	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/babe"
+	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/lib/common/variadic"
 	"github.com/ChainSafe/gossamer/lib/transaction"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 
 	"github.com/stretchr/testify/require"
 )
 
-// TODO: add test against latest gssmr runtime
-// See https://github.com/ChainSafe/gossamer/issues/2703
+func buildBlockWithSlotAndTimestamp(t *testing.T, instance state.Runtime,
+	parent *types.Header, currentSlot, timestamp uint64) *types.Block {
+	t.Helper()
+
+	digest := types.NewDigest()
+	prd, err := types.NewBabeSecondaryPlainPreDigest(0, currentSlot).ToPreRuntimeDigest()
+	require.NoError(t, err)
+	err = digest.Add(*prd)
+	require.NoError(t, err)
+	header := &types.Header{
+		ParentHash:     parent.Hash(),
+		StateRoot:      common.Hash{},
+		ExtrinsicsRoot: common.Hash{},
+		Number:         parent.Number + 1,
+		Digest:         digest,
+	}
+
+	err = instance.InitializeBlock(header)
+	require.NoError(t, err)
+
+	idata := types.NewInherentData()
+	err = idata.SetInherent(types.Timstap0, timestamp)
+	require.NoError(t, err)
+
+	err = idata.SetInherent(types.Babeslot, currentSlot)
+	require.NoError(t, err)
+
+	parachainInherent := babe.ParachainInherentData{
+		ParentHeader: *parent,
+	}
+
+	err = idata.SetInherent(types.Parachn0, parachainInherent)
+	require.NoError(t, err)
+
+	err = idata.SetInherent(types.Newheads, []byte{0})
+	require.NoError(t, err)
+
+	ienc, err := idata.Encode()
+	require.NoError(t, err)
+
+	// Call BlockBuilder_inherent_extrinsics which returns the inherents as encoded extrinsics
+	inherentExts, err := instance.InherentExtrinsics(ienc)
+	require.NoError(t, err)
+
+	// decode inherent extrinsics
+	var inExts [][]byte
```

Comment:
nit no need for comment if the var is named explicitly imo

```suggestion
	var inherentExtrinsics [][]byte
```